src/test/java/org/prebid/server/bidder/grid/GridBidderTest.java

```diff
@@ -103,10 +202,24 @@ public void makeHttpRequestsShouldReturnErrorIfExtImpGridZero() {
         final Result<List<HttpRequest<BidRequest>>> result = gridBidder.makeHttpRequests(bidRequest);
 
         // then
-        assertThat(result.getErrors()).hasSize(1);
-        assertThat(result.getErrors().get(0).getMessage()).startsWith("uid is empty");
-        assertThat(result.getErrors().get(0).getType()).isEqualTo(BidderError.Type.bad_input);
         assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors()).contains(BidderError.badInput("Empty uid in imp with id: 123"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorOnZeroValidImps() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(emptyList())
+                .id("request_id")
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = gridBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors()).contains(BidderError.badInput("No valid impressions for grid"));
```

Comment:
containsExactly is not suitable here, because when there is no valid imps in request, bidder adds an error:
`errors.add(BidderError.badInput("No valid impressions for grid"));`

so, whenever i use contains instead of containsExactly, i try to check for existence of only this error.